### ISSUE TITLE
include >=v1.2 structs in SectionOrdering rule

### ISSUE THREAD

**Issue by a-frantz** [stjude-rust-labs/wdl#202]
The [`SectionOrdering` lint rule](https://github.com/stjude-rust-labs/wdl/blob/main/wdl-lint/src/rules/section_order.rs) should be updated to account for the optional `meta` and `parameter_meta` sections which were introduced to `struct` definitions in [v1.2 of the WDL specification](https://github.com/openwdl/wdl/blob/wdl-1.2/SPEC.md#struct-definition).

The order should be `meta -> parameter_meta -> declarations`.

**Comment by bhartiRaz** [comment]
"Hello,  @a-frantz  could you please assign this issue to me?"

**Comment by a-frantz** [comment]
Sure thing! @bhartiRaz

**Comment by bhartiRaz** [comment]
Thanks, @a-frantz! I'll start working on this.

**Comment by bhartiRaz** [comment]
Hello @a-frantz , I am a little new to the code base, could you please clarify where to look for the code, and elaborate further? Thankyou!

**Comment by claymcleod** [comment]
I agree that this issue is not clear on what needs to happen. @a-frantz, please add more details.

When it does come time to implement it, you'll be looking at this file @bhartiRaz: https://github.com/stjude-rust-labs/wdl/blob/main/wdl-lint/src/rules/section_order.rs.

**Comment by a-frantz** [comment]
@bhartiRaz I've updated the issue to include some more detail.

You'll need to edit the `section_order.rs` file linked to above and also the [`section-ordering/source.wdl`](https://github.com/stjude-rust-labs/wdl/blob/main/wdl-lint/tests/lints/section-ordering/source.wdl) file to include example `struct` definitions which both do and do not trigger the modified lint rule.

Check the [FAQ section of the CONTRIBUTING guide](https://github.com/stjude-rust-labs/wdl/blob/main/CONTRIBUTING.md) for some more direction. And you can see the linked section of the WDL spec to see how structs should be defined in the example WDL.

**Comment by dead8309** [comment]
Hi @claymcleod @a-frantz looks like this issue has had no progress for past 3weeks. Can you assign me this?

**Comment by a-frantz** [comment]
@dead8309 it's all yours!
